File: pureboard-web/src/main/java/org/pureboard/dto/ContenuDto.java
package org.pureboard.dto;

import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.Data;

import java.util.List;

@JsonInclude(JsonInclude.Include.NON_NULL)
@Data
public class ContenuDto {

    private TypeContenu type;

    private String texte;

    private TableauDto tableau;

    private String classe;

    private String lien;

    private List<ContenuDto> listeContenu;

    private String nomMethode;

    private String parametresMethode;
}

File: pureboard-jquery/src/entity/contenu-dto.ts
import {TypeContenuDto} from './type-contenu-dto';
import {TableauDto} from './tableau-dto';

export class ContenuDto {

  type: string = TypeContenuDto.INCONNU;
  texte: string = '';
  tableau: TableauDto | null = null;

  classe: string = '';

  lien: string = '';

  listeContenu: ContenuDto[] = [];

  nomMethode: string = '';

  parametresMethode: string = '';
}
